### PR TITLE
macos support

### PR DESCRIPTION
**Pull request by ccqpein**
+ add foreign libraries names
+ give cc-flags value

### CODE REVIEW

File: src/grovel.lisp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 (in-package :jpeg-turbo)
 (include "turbojpeg.h")
-#+freebsd
+#+(or freebsd OS-MACOSX)
```
Comment (ccqpein) [inline src/grovel.lisp:3]: For macos (feature is `OS-MACOSX`), global include path also is `/usr/local/include`.

**Comment by ccqpein** [comment]
Two changes in this PR:

1. For macos (fthe eature is OS-MACOSX), global include path also is /usr/local/include.
2. `.so` files are `.dylib` in macos.

For macOS users (if someone find this PR in future) who use `homebrew` install `jpeg-turbo`. You need to do these first:

```shell
ln -sfv /usr/local/Cellar/jpeg-turbo/2.0.4/lib/libturbojpeg.dylib /usr/local/lib/libturbojpeg.dylib
ln -sfv /usr/local/Cellar/jpeg-turbo/2.0.4/include/turbojpeg.h  /usr/local/include/turbojpeg.h
```

Now `jpeg-turbo` load well on my computer.

File: src/jpeg-turbo.lisp [hunk 2]
```diff
@@ -12,8 +12,10 @@
 
 ;; Library
 (define-foreign-library jpeg-turbo
+  (:os-macosx (:or "libturbojpeg.0.dylib" "libturbojpeg.dylib"))
```
Comment (ccqpein) [inline src/jpeg-turbo.lisp:15]: Depend on [cffi doc](https://common-lisp.net/project/cffi/manual/html_node/define_002dforeign_002dlibrary.html), `load the library name in order for the first one where feature evaluates to true`, so I put `os-macosx` before `unix`.

**Comment by shamazmazum** [comment]
Thanks!

**Pull request merged**